### PR TITLE
Hot Feature/export leads

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on: https://github.com/toggle-corp/react-store/pull/604
Depends on: https://github.com/the-deep/server/pull/342

Fix https://github.com/the-deep/deeper/issues/706
Fix https://github.com/the-deep/deeper/issues/707
Fix https://github.com/the-deep/deeper/issues/710
Fix https://github.com/the-deep/deeper/issues/711
Fix https://github.com/the-deep/deeper/issues/712

### CODE REVIEW

File: src/redux/selectors/siloDomainData/addLeads.js [hunk 1]
```diff
@@ -217,6 +221,12 @@ export const addLeadViewCompletedLeadKeysSelector = createSelector(
     addLeadViewCompletedLeadsSelector,
     leads => leads.map(leadAccessor.getKey),
 );
+
+export const addLeadViewCompletedServerIdsSelector = createSelector(
+    addLeadViewCompletedLeadsSelector,
+    leads => leads.map(leadAccessor.getServerId),
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/addLeads.js]: Don't we need to filter `null` or `undefined` values?

File: src/views/Leads/ActionButtons/index.js [hunk 2]
```diff
@@ -101,6 +105,9 @@ export default class ActionButtons extends React.PureComponent {
         };
     }
 
+    // NOTE: This is sent as an array as LeadCopyModal is built for bulk operations
+    getId = memoize(row => [row.id]);
```
Comment (tnagorra) [inline src/views/Leads/ActionButtons/index.js]: We can still name this method: `getSomethingIds`

Comment (tnagorra) [reply src/views/Leads/ActionButtons/index.js]: `getLeadIds`

File: src/views/Leads/GridItem/index.js [hunk 3]
```diff
@@ -99,6 +104,9 @@ export default class GridItem extends React.PureComponent {
         return icon;
     }
 
+    // NOTE: This is sent as an array as LeadCopyModal is built for bulk operations
+    getId = memoize(row => [row.id]);
```
Comment (tnagorra) [inline src/views/Leads/GridItem/index.js]: The function name could be more informative.

**Review by tnagorra — CHANGES_REQUESTED** [review]
Code looks almost okay!

File: src/components/general/LeadCopyModal/index.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        url: '/leads-copy/',
+        body: ({ params: { body } }) => body,
+        method: requestMethods.POST,
+        onSuccess: ({ response: { leads, projects }, props: { closeModal } }) => {
+            const message = _ts(
+                'leads.copyModal',
+                'successNotify',
+                {
+                    leadsCount: leads.count,
+                    projectsCount: projects.count,
```
Comment (thenav56) [inline src/components/general/LeadCopyModal/index.js]: `.count` Is this new in JS?

File: src/components/general/LeadCopyModal/index.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                message: _ts('leads.copyModal', 'leadsCopyFailureMessage'),
+                duration: notify.duration.MEDIUM,
+            });
+        },
+        schemaName: 'leadsCopy',
+    },
+};
+
+@connect(mapStateToProps)
+@RequestCoordinator
```
Comment (tnagorra) [inline src/components/general/LeadCopyModal/index.js]: We generally do not need request coordinator in a Modal.

File: src/redux/selectors/siloDomainData/addLeads.js [hunk 6]
```diff
@@ -217,6 +222,12 @@ export const addLeadViewCompletedLeadKeysSelector = createSelector(
     addLeadViewCompletedLeadsSelector,
     leads => leads.map(leadAccessor.getKey),
 );
+
+export const addLeadViewCompletedServerIdsSelector = createSelector(
+    addLeadViewCompletedLeadsSelector,
+    leads => leads.map(leadAccessor.getServerId).map(isDefined),
```
Comment (tnagorra) [inline src/redux/selectors/siloDomainData/addLeads.js]: I think this should be filter instead of map

**Pull request merged**